### PR TITLE
gopass-jsonapi 1.11.1 (new formula)

### PR DESCRIPTION
**Pull request by giffels**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`gopass` developers decided to move `gopass-jsonapi` needed for the browser integration of `gopass` to a separate package. This pull request will add a formula for this package in order to use `gopass` browser integration.

### CODE REVIEW

**Comment by giffels** [comment]
GitHub Actions CI is failing due to the GitHub repository is not notable enough. As I have written above, the developers recently decided to move `gopass-jsonapi` to separate project. So, the number of forks is still quite low. However, the mother project `gopass` is much more notable (see https://github.com/gopasspw/gopass).

File: Formula/gopass-jsonapi.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  depends_on "go" => :build
+  depends_on "gopass"
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags", "-s -w -X main.version=#{version}"
+  end
+
+  def caveats
+    "You need to run gopass-jsonapi configure for each browser you want to use with gopass."
+  end
```
Comment (carlocab) [inline Formula/gopass-jsonapi.rb:17]: This is already documented upstream, so there's no need to include this. `caveats` are for Homebrew-specific messages.

Comment (giffels) [reply Formula/gopass-jsonapi.rb]: Thanks. I have removed the `caveats` block.

**Comment by MikeMcQuaid** [comment]
> `gopass` developers decided to move `gopass-jsonapi` needed for the browser integration of `gopass` to a separate package. This pull request will add a formula for this package in order to use `gopass` browser integration.

This is fine with me 👍🏻

File: Formula/gopass-jsonapi.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ensure
+      system Formula["gnupg"].opt_bin/"gpgconf", "--kill", "gpg-agent"
+      system Formula["gnupg"].opt_bin/"gpgconf", "--homedir", "keyrings/live",
+                                                 "--kill", "gpg-agent"
+    end
+
+    assert_match version.to_s, shell_output("#{bin}/gopass-jsonapi --version")
+
+    msg='{"type": "query", "query": "foo.bar"}'
+    out, _err, _status = Open3.capture3("#{bin}/gopass-jsonapi listen", stdin_data: [msg.length].pack("L<")+msg)
```
[the remark this answers is not in the document]
Comment (giffels) [reply Formula/gopass-jsonapi.rb]: You mean like in the following code?

```ruby
out = pipe_output("#{bin}/gopass-jsonapi listen", [msg.length].pack("L<")+msg)
```

That works indeed. I have adjusted the code accordingly. Thanks for the hint!

File: Formula/gopass-jsonapi.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      system Formula["gnupg"].opt_bin/"gpgconf", "--kill", "gpg-agent"
+      system Formula["gnupg"].opt_bin/"gpgconf", "--homedir", "keyrings/live",
+                                                 "--kill", "gpg-agent"
+    end
+
+    assert_match version.to_s, shell_output("#{bin}/gopass-jsonapi --version")
+
+    msg='{"type": "query", "query": "foo.bar"}'
+    out = pipe_output("#{bin}/gopass-jsonapi listen", [msg.length].pack("L<")+msg)
+    assert_match %r{Email/other@foo.bar}, out
```
Comment (carlocab) [inline Formula/gopass-jsonapi.rb:43]:
```suggestion
    assert_match "Email/other@foo.bar", out
```
You don't use any regex tokens here, so I think you can just pass this as a string. Squash your commits together if you make this change, please. Thanks!

Comment (giffels) [reply Formula/gopass-jsonapi.rb]: Thanks. I have removed the regex token and squashed my commits.

**Review by carlocab — DISMISSED** [review]

File: Formula/gopass-jsonapi.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+      system Formula["gopass"].opt_bin/"gopass", "init", "--path", testpath, "noop", "testing@foo.bar"
+      system Formula["gopass"].opt_bin/"gopass", "generate", "Email/other@foo.bar", "15"
+    ensure
+      system Formula["gnupg"].opt_bin/"gpgconf", "--kill", "gpg-agent"
+      system Formula["gnupg"].opt_bin/"gpgconf", "--homedir", "keyrings/live",
+                                                 "--kill", "gpg-agent"
+    end
+
+    assert_match version.to_s, shell_output("#{bin}/gopass-jsonapi --version")
```
Comment (MikeMcQuaid) [inline Formula/gopass-jsonapi.rb:39]:
```suggestion
    assert_match /^#{version}$/, shell_output("#{bin}/gopass-jsonapi --version")
```
or a similar regex that works to ensure e.g. `1.11.1` won't match `21.11.12`

File: Formula/gopass-jsonapi.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    msg='{"type": "query", "query": "foo.bar"}'
+    out = pipe_output("#{bin}/gopass-jsonapi listen", [msg.length].pack("L<")+msg)
+    assert_match "Email/other@foo.bar", out
```
Comment (MikeMcQuaid) [inline Formula/gopass-jsonapi.rb:43]:
```suggestion
    msg = '{"type": "query", "query": "foo.bar"}'
    assert_match "Email/other@foo.bar", 
      pipe_output("#{bin}/gopass-jsonapi listen", "#{[msg.length].pack("L<")}#{msg}")
```

**Comment by giffels** [comment]
Thanks @MikeMcQuaid, I have applied the requested changes.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @giffels!

**Pull request closed without merging**